From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Mon, 15 Apr 2019 18:26:33 -0400
Subject: wait builtin: avoid hanging on inherited children

in https://lists.gnu.org/archive/html/bug-bash/2019-04/msg00096.html,
Chet Ramey proposes this fix to avoid the wait builtin hanging on
previously unknown children.

This addresses debian bug #920455.
---
 jobs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jobs.c b/jobs.c
index ce2bdf2..2c24537 100644
--- a/jobs.c
+++ b/jobs.c
@@ -2488,7 +2488,7 @@ wait_for_background_pids ()
     r = wait_for (last_procsub_child->pid);
   wait_procsubs ();
   reap_procsubs ();
-#if 1
+#if 0
   /* We don't want to wait indefinitely if we have stopped children. */
   /* XXX - should add a loop that goes through the list of process
      substitutions and waits for each proc in turn before this code. */
